internal/crypto/crypto_test.go

```diff
@@ -1,29 +1,53 @@
-package crypto_test
+package crypto
```

Comment:
Hmm according to the manifesto, this should be crypto_test (same for other tests in other commits)

Reply:
I would have to read a little about it to be able to define precisely how Go handles test/classic dependencies, but in any case if you want to use (correctly) the Go SDK, you need to download all its dependencies, including test ones.
Could you please elaborate on why you find new tests harder to read? I find the indentation is better, the output more concise (and clearer, e.g. with `go test ./...`), and the possibility to vectorize things a better option.
You can also see that we don't test internal workings, which is IMHO a good thing, along with the fact that this is more common in the Go community.
Finally, `ginkgo` and `gomega` imply huge machinery and logical concepts that are quite heavy, that is, not needed here.

Nonetheless, if you have suggestions concerning how I implemented the tests, I'm of course all ears and can still change that.

As for the package name, that is true and I'm gonna change it, thx for reporting it! :)

Reply:
It's mostly all the `if err != nil { t.fatal("abort") }` boilerplate that I don't like.
And the gomega style of having one line per assertion instead of `if err != nil || id == nil || *id == "" {`.
Also, if a test fails, `err` is not printed, only your message is.

The vectorization part is fine, but it could be done with gomega too I think.

I won't be working much on this package anyway, so this is fine by me, just my 2 cents ^^

Reply:
I'm gonna try and improve logging on error. I've also seen many "want %v ; got %v" in the std lib, that could do the trick for errors.
Yes indeed vectorization can be done with gomega (or is it ginkgo? I never remember), that's my bad.

As for error handling, apart from putting this into a helper function, I think this has more to do with Go than anything else. 

I like Go's `testing` framework because it is simple and plain.
We can skip the "test suite" part with this and such. Also the `go test` command understands all semantics of the testing package, where you need to know additional flags and other mechanisms to interact well with third-party libs.

Anyway the idea is not (at all) to have a SDK that repulses you or others so if you feel that's needed, we can discuss those points. Just let me know!